sale_commission/models/account_invoice.py

```diff
@@ -118,15 +118,19 @@ class AccountInvoiceLineAgent(models.Model):
         compute="_compute_settled",
         store=True, copy=False)
 
-    @api.depends('object_id.price_subtotal')
+    @api.depends('commission', 'commission.amount_base_type',
+                 'commission.commission_type', 'commission.fix_qty',
+                 'invoice.type', 'object_id.commission_free',
+                 'object_id.price_subtotal', 'object_id.price_unit',
+                 'object_id.quantity')
```

Comment:
Why do we need here `object_id.quantity`?

In other hand, changing commissions type shouldn't modify retroactively all the commissions amount already computed, so you should remove that from the depends

Reply:
Thanks @pedrobaeza for reviewing.
Here I added all the fields that are involved in the computation of `amount` (that is, all the field involved in this method) because changing any of them might change the value of `amount`.
Changing `object_id.quantity` might change computed value of `amount` so it should trigger a recomputation.

If the recomputation of `amount` shouldn't be done under some circumstances, those circumstances should be checked inside the method to prevent the recomputation.

Reply:
> Changing object_id.quantity might change computed value of amount so it should trigger a recomputation.

If quantity changes the amount, then the trigger will be risen any way, so don't add extra burden to the compute method.

> If the recomputation of amount shouldn't be done under some circumstances, those circumstances should be checked inside the method to prevent the recomputation.

No, you shouldn't check anything inside the method. You simply shouldn't trigger the compute method for that data. As said, you don't retroactively to recompute every past amount when you change the commission rate or similar. That's simply out of business logic.

Reply:
> the trigger will be risen any way

How will it be risen if the method `_compute_amount` does not _depend_ on `object_id.quantity`?

> don't retroactively to recompute every past amount when you change the commission rate or similar

I agree, so maybe inside the `_compute_amount` we should execute the computation of `amount` only for the lines (`account.invoice.line.agent` records) that have not _already been computed_, maybe checking the `settled` field?

Reply:
> How will it be risen if the method _compute_amount does not depend on object_id.quantity?

Because it depends on `object_id.price_subtotal`, so it's a transitive effect:

object_id.quantity > object_id.price_subtotal > amount.

> I agree, so maybe inside the _compute_amount we should execute the computation of amount only for the lines (account.invoice.line.agent records) that have not already been computed, maybe checking the settled field?

But that's an specific use case, but I don't want the change to be applied even if they are not settled in my use cases. This should be applied from now on when you do the change. If you want to recompute it, do it with an script, or add a tool for recomputing amounts. And not only this:

- User will experiment a freeze on Odoo saving this value without knowing what is happening.
- This change is too much behavior change for an stable version (but I repeat, I won't do it even on new version due to the stated reasons).

Reply:
ok @pedrobaeza, I restored the previous behavior in https://github.com/OCA/commission/pull/192/commits/ed661de4e59aa18c6a0d1b3ed9a7218a385120b9